client/components/focusable/index.jsx

```diff
@@ -2,24 +2,21 @@
 /**
  * External dependencies
  */
-import React, { Component } from 'react';
 import classNames from 'classnames';
-import { omit } from 'lodash';
 import PropTypes from 'prop-types';
+import React, { Component } from 'react';
+import { noop } from 'lodash';
 
 class Focusable extends Component {
 	static propTypes = {
 		onClick: PropTypes.func.isRequired,
 		onKeyDown: PropTypes.func,
 	};
 
-	onClick = event => {
-		const { onClick } = this.props;
-		onClick( event );
-	};
+	static defaultProps = { onClick: noop };
```

Comment:
**This was the main motivation for this PR**

@ryelle It may go against the spirit of `Focusable`, but what if we want to let the event bubble and be handled by a parent?

This ensures we don't error in that case.

Reply:
Do you have an example of where/why you might want to do this? I'm not clear on how this would be used, so I can't think through any possible a11y implications.

Reply:
My initial attempt at fixing the a11y error on this `<a href="#">`:

https://github.com/Automattic/wp-calypso/blob/9de5fde73763a791255f9426fde128d41c89a907/client/me/purchases/remove-purchase/index.jsx#L237-L242

Was to simply replace the `<a href="#">` with `<Focusable>`. This was an error if you `onClick` was not provided to the Focusable. This naive replacement would allow the events to bubble up to the Card where the event is handled.

I ended up finding what I believe is a better approach (below), however it doesn't seem unreasonable to allow events _not to be handled_ by the `Focusable` and instead let them bubble up and be handled by an ascendant.

https://github.com/Automattic/wp-calypso/blob/2316a2592e5193382c00e405a0ede9186e5f1e99/client/me/purchases/remove-purchase/index.jsx#L467-L468

Reply:
I'm still not sure I understand what scenario you'd use this in. Using `<a href"#" />` like that was an anti-pattern (which the linter caught) - the onClick should be on the clickable element itself.

Considering `Focusable` should be [a "last resort" component](https://wpcalypso.wordpress.com/devdocs/design/focusable), I don't think we should add this functionality to it.

Reply:
Sounds like we'd like to ensure that `onClick` _is indeed provided_. I've rolled back the provided default, but kept safety by protecting against calling a nonexistent `onClick` function.